pkg/ipcache/ipcache.go

```diff
@@ -90,33 +90,71 @@ func (ipc *IPCache) RUnlock() {
 	ipc.mutex.RUnlock()
 }
 
-// Upsert adds / updates the provided IP and identity into both caches contained
-// within ipc.
-func (ipc *IPCache) Upsert(endpointIP string, identity identity.NumericIdentity) {
+// UpsertIP updates / inserts the provided IP->Identity mapping into the
+// kvstore, which will subsequently trigger an event in ipIdentityWatcher().
+func UpsertIP(IP net.IP, ID identity.NumericIdentity, metadata string) error {
```

Comment:
funny, in the full PR I ended up doing this anyway, but it wasn't part of the regular refactoring. I'll do this.

Soon, it might be worth splitting the ipcache kvstore push stuff from the in-memory / kvstore event handling piece. That would clarify things a bit.